test/Test.hs

```diff
@@ -26,14 +25,14 @@ runTest labels (Test name expect property) = do
     let label = "[" ++ intercalate "." (reverse labels) ++ "] " ++ name
     result <- quickCheckWithResult (stdArgs { chatty = False }) property
     case (expect, isSuccess result) of
-        (ExpectSuccess, True)  -> putStrLn $ "OK: " ++ label
-        (ExpectFailure, False) -> putStrLn $ "OK (expected failure): " ++ label
+        (ExpectSuccess, True) ->
+            putStrLn $ "[OK] " ++ label
+        (ExpectFailure, False) ->
+            putStrLn $ "[OK, expected failure] " ++ label
+        (ExpectFailure, True) ->
+            putStrLn $ "[Warning, unexpected success] " ++ label
```

Comment:
Someone unfamiliar with this test runner may rely on this to fail.

You could add a `UncheckedCounterexample` constructor to `Expect`, to hint to the reader that it's not suitable for checking that something is always `False`. You can also print a more informative message when it happens.